src/main/java/com/dih/connector/test/service/ConsumeOfferService.java

```diff
@@ -0,0 +1,126 @@
+package com.dih.connector.test.service;
+
+import com.dih.connector.test.client.connector.model.AgreementResponse;
+import com.fasterxml.jackson.core.util.DefaultIndenter;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.client.support.BasicAuthenticationInterceptor;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import javax.annotation.PostConstruct;
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+import java.util.UUID;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ConsumeOfferService {
+
+    @Value("${consumer.baseUrl}")
+    private URI consumerBaseUrl;
+
+    @Value("${connector.baseUrl}")
+    private URI producerBaseUrl;
+
+    @Qualifier("json-ld")
+    private final RestTemplate restTemplateLd;
+
+    @Qualifier("json-default")
+    private final RestTemplate restTemplateDefault;
+
+    private ObjectMapper objectMapper;
+
+    @PostConstruct
+    public void init() {
+        objectMapper = new ObjectMapper();
```

Comment:
Maybe we can move ObjectMapper bean creation into configuration?

Reply:
it is not a system-wide one, used for debugging purpose only. To pretty print the messages.